src/Cake.Core/IO/ProcessRunner.cs

```diff
@@ -69,18 +69,19 @@ public IProcess Start(FilePath filePath, ProcessSettings settings)
 
             // Get the working directory.
             var workingDirectory = settings.WorkingDirectory ?? _environment.WorkingDirectory;
-            settings.WorkingDirectory = workingDirectory.MakeAbsolute(_environment);
 
             // Create the process start info.
             var info = new ProcessStartInfo(fileName)
             {
                 Arguments = arguments.Render(),
-                WorkingDirectory = workingDirectory.FullPath,
+                WorkingDirectory = settings.WorkingDirectory != null ? workingDirectory.FullPath : null,
                 UseShellExecute = false,
                 RedirectStandardError = settings.RedirectStandardError,
                 RedirectStandardOutput = settings.RedirectStandardOutput
             };
 
+            settings.WorkingDirectory = workingDirectory.MakeAbsolute(_environment);
```

Comment:
Doesn't reallty make sense to set this after it's been assigned to the `ProcessStartInfo`. This could be removed.